src/proc/occlusion-filter.cpp

```diff
@@ -45,51 +71,111 @@ namespace librealsense
     // -  The occlusion is designated as U coordinate for a given pixel is less than the U coordinate of the predecessing pixel.
     // -  The UV mapping for the occluded pixel is reset to (0,0). Later on the (0,0) coordinate in the texture map is overwritten
     //    with a invalidation color such as black/magenta according to the purpose (production/debugging)
-    void occlusion_filter::monotonic_heuristic_invalidation(float3* points, float2* uv_map, const std::vector<float2> & pix_coord) const
-    {
-        float occZTh = 0.1f; //meters
-        int occDilationSz = 1;
-        auto pixels_ptr = pix_coord.data();
-        auto points_width = _depth_intrinsics->width;
-        auto points_height = _depth_intrinsics->height;
-
-        for (size_t y = 0; y < points_height; ++y)
-        {
-            float maxInLine = -1;
-            float maxZ = 0;
-            int occDilationLeft = 0;
-
-            for (size_t x = 0; x < points_width; ++x)
-            {
-                if (points->z)
-                {
-                    // Occlusion detection
-                    if (pixels_ptr->x < maxInLine || (pixels_ptr->x == maxInLine && (points->z - maxZ) > occZTh))
-                    {
-                        uv_map->x = 0.f;
-                        uv_map->y = 0.f;
-                        occDilationLeft = occDilationSz;
-                    }
-                    else
-                    {
-                        maxInLine = pixels_ptr->x;
-                        maxZ = points->z;
-                        if (occDilationLeft > 0)
-                        {
-                            uv_map->x = 0.f;
-                            uv_map->y = 0.f;
-                            occDilationLeft--;
-                        }
-                    }
-                }
-
-                ++points;
-                ++uv_map;
-                ++pixels_ptr;
-            }
-        }
-    }
-
+   void occlusion_filter::monotonic_heuristic_invalidation(float3* points, float2* uv_map, const std::vector<float2>& pix_coord, const rs2::depth_frame& depth) const
+   {
+       float occZTh = 0.1f; //meters
+       int occDilationSz = 1;
+       auto points_width = _depth_intrinsics->width;
+       auto points_height = _depth_intrinsics->height;
+       auto pixels_ptr = pix_coord.data();
+       auto points_ptr = points;
+       auto uv_map_ptr = uv_map;
+       float maxInLine = -1;
+       float maxZ = 0;
+
+       if (_occlusion_scanning == horizontal)
+       {
+
+           {
+               scoped_timer t1("Horizontal Scan");
```

Comment:
@gwen2018 - please remove the two scoped_timers from the CPU version

Reply:
removed.